python/services/evaluationservice/dmod/evaluationservice/__main__.py

```diff
@@ -0,0 +1,130 @@
+
+    When proceeding to check environment variables, the following variables are checked in the following order, with the
+    first existing and non-``None`` value returned:
+
+        - DOCKER_SECRET_REDIS_<SUFFIX>
+        - REDIS_<SUFFIX>
+        - DOCKER_REDIS_<SUFFIX>
+
+    The actual names have ``<SUFFIX>`` replaced with the provided suffix parameter.
+
+    Finally, the fallback value is returned when necessary.
+
+    Parameters
+    ----------
+    parsed_val
+    env_var_suffix
+    fallback
+
+    Returns
+    -------
+    The appropriate value, given the params and the existing environment variables.
+    """
+    if parsed_val is not None:
+        return parsed_val
+    env_prefixes = ['REDIS_', 'DOCKER_SECRET_REDIS_', 'DOCKER_REDIS_']
+    for prefix in env_prefixes:
+        env_var = prefix + env_var_suffix
+        if getenv(env_var, None) is not None:
+            return getenv(env_var)
+    return fallback
+
+
+def redis_params(parsed_args) -> tuple:
+    """
+    Return the Redis host, port, and password for connections, obtained from the parsed arguments and/or environment.
+
+    Parameters
+    ----------
+    parsed_args
+
+    Returns
+    -------
+    tuple
+        The Redis host, port, and password for connections.
+    """
+    host = _get_parsed_or_env_val(parsed_args.redis_host, 'HOST', 'localhost')
+    port = _get_parsed_or_env_val(parsed_args.redis_port, 'PORT', 6379)
+    passwd = _get_parsed_or_env_val(parsed_args.redis_pass, 'PASS', '')
+
+    return host, port, passwd
+
+
+def main():
+    args = _handle_args()
+    #TODO add args to allow different service definition,
+    #i.e. dev test
+    #if args.dev:
+    #   run_dev_stuff()
+    #else: run_prod()
+    resource_list = read_resource_list(Path(args.resource_list_file))
+
+    # Obtain Redis params
+    redis_host, redis_port, redis_pass = redis_params(args)
+
+    # instantiate the resource manager for the scheduler
+    resource_manager = RedisManager("evaluations", redis_host=redis_host, redis_port=redis_port, redis_pass=redis_pass)
+    resource_manager.set_resources(resource_list)
+
+    # instantiate the scheduler
+    # TODO: look at handling if the value in args.images_and_domains_yaml doesn't correspond to an actual file
+    launcher = Launcher(images_and_domains_yaml=args.images_and_domains_yaml, type="dev")
+    # instantiate the job manager
+    job_manager: JobManager = JobManagerFactory.factory_create(resource_manager, launcher, host=redis_host, port=redis_port, redis_pass=redis_pass)
+
+    #Instansite the handle_job_request
+    handler = SchedulerHandler(job_manager, ssl_dir=Path(args.ssl_dir), port=args.port)
+    # Create the async task for processing Jobs within queue and scheduling
+    handler.add_async_task(job_manager.manage_job_processing())
+    #keynamehelper.set_prefix("stack0")
+    handler.run()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
It looks like this file was copied from an old version of the schedulerservice package.  If it stays in something close to its current form, it will need considerable modification.

Reply:
Yeah, that's a bad copy from when I was first stitching everything together, pre-django. I thought that got deleted. Removing.... now.